### PR TITLE
Fix listing all service principals

### LINKED ISSUE #1481: ad_serviceprincipal_info does not properly list all service principals
##### SUMMARY

ad_serviceprincipal_info does not properly list all service principals when neither object_id nor app_id are specified. This is due to the filter incorrectly including the app_id for every invocation. When app_id is `None`, the results are always blank.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

azure_rm_adserviceprincipal_info

##### ANSIBLE VERSION

```paste below
ansible [core 2.11.12]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/kmcdonough/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = <redacted>
  ansible collection location = /home/kmcdonough/.ansible/collections:/usr/share/ansible/collections
  executable location = <redacted>
  python version = 3.8.10 (default, Nov 22 2023, 10:22:35) [GCC 9.4.0]
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

```paste below
2.1.1
```

##### CONFIGURATION

```paste below
n/a
```

##### OS / ENVIRONMENT

```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
```

##### STEPS TO REPRODUCE

Invoke the module with no arguments

```yaml
        - name: get all application service principals
          azure.azcollection.azure_rm_adserviceprincipal_info:
          register: all_service_principals
```

##### EXPECTED RESULTS

The registered variable should contain a list of all service principals in the tenant

##### ACTUAL RESULTS

The module always returns an empty list, as the filter is misconfigured by default

```paste below
ok: [abbpsd-dev-strawman] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "app_id": null,
            "auth_source": "auto",
            "cert_validation_mode": null,
            "client_id": null,
            "cloud_environment": "AzureCloud",
            "log_mode": null,
            "log_path": null,
            "object_id": null,
            "password": null,
            "profile": null,
            "secret": null,
            "subscription_id": null,
            "tenant": null,
            "thumbprint": null,
            "x509_certificate_path": null
        }
    },
    "service_principals": [] <-- this is incorrect, the tenant has many service principals
}
```

### PR DESCRIPTION
**Pull request by kent007**
##### SUMMARY

Fixes https://github.com/ansible-collections/azure/issues/1481

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

azure_rm_adserviceprincipal_info

##### ADDITIONAL INFORMATION

I added reply pagination and also corrected the documentation to specify the results come back in a container

### CODE REVIEW

File: plugins/modules/azure_rm_adserviceprincipal_info.py [hunk 1]
```diff
@@ -43,30 +43,34 @@
...
-    returned: always
-    sample: false
-object_id:
-    description:
-        - It's service principal's object ID.
-    returned: always
-    type: str
-    sample: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+service_principals:
+    type: list
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_adserviceprincipal_info.py:52]: Missing 'description'

File: plugins/modules/azure_rm_adserviceprincipal_info.py [hunk 2]
```diff
@@ -43,30 +43,34 @@
...
+    elements: dict
+    contains:
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_adserviceprincipal_info.py:55]: Missing 'returned' definde!

**Review by Fred-sun — COMMENTED** [review]
Small change qurest!

File: plugins/modules/azure_rm_adserviceprincipal_info.py (hunk 1 above)
Comment (kent007) [reply plugins/modules/azure_rm_adserviceprincipal_info.py:52]: Updated

File: plugins/modules/azure_rm_adserviceprincipal_info.py (hunk 2 above)
Comment (kent007) [reply plugins/modules/azure_rm_adserviceprincipal_info.py:55]: Updated

**Comment by kent007** [comment]
I think comments should be addressed. I also included a new example

File: plugins/modules/azure_rm_adserviceprincipal_info.py [hunk 3]
```diff
@@ -40,33 +40,44 @@
 - name: get ad sp info
   azure_rm_adserviceprincipal_info:
     app_id: "{{ app_id }}"
+
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_adserviceprincipal_info.py]: Trailing spaces
```suggestion

```

**Review by Fred-sun — COMMENTED** [review]
small change request!

File: plugins/modules/azure_rm_adserviceprincipal_info.py [hunk 4]
```diff
@@ -40,33 +40,43 @@
...
-        - The application ID.
-    returned: always
-    type: str
-    sample: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
-app_role_assignment_required:
-    description:
-        - Whether the Role of the Service Principal is set.
-    type: bool
+service_principals:
+    description:
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_adserviceprincipal_info.py]: trailing whitespace
```suggestion
    description:
```

**Review by Fred-sun — COMMENTED** [review]
small change request!

**Review by xuzhang3 — APPROVED** [review]

**Review by xuzhang3 — APPROVED** [review]

**Pull request merged**